microsoftgraph/microsoft-graph-explorer-v4
# Task: cache autocomplete results

## Overview

Uses indexed db to store the results of the autocomplete query so that subsequent calls do not call the API. This effectively speeds up autocomplete triggers based on the same url

Fixes #724

### Demo

![image](https://user-images.githubusercontent.com/58787602/98675361-343fc780-236b-11eb-9ff6-020d4593ee93.png)

### Notes

Request URLs with keys are replaced with generic IDs so that they are easily recreated for storage and searching in the cache.

### Tasks

- [x] Create store for autocomplete options
- [x] Read and write options to the store
- [x] Add a time to live for the options in the store

## Testing Instructions

* Run the app. Trigger an autocomplete search e.g `me/messages?`
* Using the developer console, notice that the request URL is placed as a key in the indexedDb storage
* perform another autocomplete search e.g `me/todo?`
* Using the developer console, notice that there was a network request that was made
* Redo `me/messages?` and notice that it is no longer calling the API
